### PR TITLE
Implement tools to add tags to scenario

### PR DESCRIPTION
**Pull request by jhou1**
This PR add the `add-tags` feature the `tools/case_id_splitter.rb`. New tags can be added following this example:

```
verification-tests ➤ tools/case_id_splitter.rb add-tags --tags @gcp,@aws,@disconnected --cid OCP-22578     
Adding tag @gcp to /Users/jhou/Projects/verification-tests/features/tierN/apiserver/custom_resource.feature for test case OCP-22578
Adding tag @aws to /Users/jhou/Projects/verification-tests/features/tierN/apiserver/custom_resource.feature for test case OCP-22578
Adding tag @disconnected to /Users/jhou/Projects/verification-tests/features/tierN/apiserver/custom_resource.feature for test case OCP-22578
verification-tests ➤ tools/case_id_splitter.rb add-tags --tags @gcp,@aws,@disconnected --cid OCP-22578    
Test case OCP-22578 already has tag @gcp
Test case OCP-22578 already has tag @aws
Test case OCP-22578 already has tag @disconnected
```

Then run `git diff` you will see tags added to the corresponding *.feature*.
```
diff --git a/apiserver/custom_resource.feature b/apiserver/custom_resource.feature
index 347636f29..cec8a86a9 100644
--- a/apiserver/custom_resource.feature
+++ b/apiserver/custom_resource.feature
@@ -2,6 +2,9 @@ Feature: API server related custom resource features

   # @author kewang@redhat.com
   # @case_id OCP-22578
+  @disconnected
+  @aws
+  @gcp
   Scenario Outline: Explain for CRD of config.openshift.io
     When I run the :explain client command with:
       | resource    | <item>                 |
```

The reason for having this feature is we want to use `tools/polarshift.rb query-cases` to select proper test cases matching a test profile, and then batch adding relevant tags to the scenarios. This implements [OCPQE-5126](https://issues.redhat.com/browse/OCPQE-5126)

@liangxia @pruan-rht @JianLi-RH @dis016 PTAL

### CODE REVIEW

**Comment by jhou1** [comment]
Until this gets /lgtm from @liangxia and @pruan-rht 
/hold

**Comment by pruan-rht** [comment]
@jhou1 have you checked this work for `Scenario Outline` feature files?

**Comment by pruan-rht** [comment]
@jhou1 good start, but it doesn't look like it works for Scenario Outlines feature files, and if I were to run the command twice, it will put tags there twice as well.
```tools/case_id_splitter.rb add-tags --tags @gcp,@aws,@disconnected --cid OCP-12049
    Examples:
      | path     | file                           | pod_name                  | expr1             | expr2                       |
      | ocp11754 | pod-request-limit-valid-3.yaml | pod-request-limit-valid-3 | cpu\\s+100m\\s+30 | memory\\s+(134217728\|128Mi)\\s+16Gi | # @case_id OCP-11754
  @disconnected
  @aws
  @gcp
  @disconnected
  @aws
  @gcp
      | ocp12049 | pod-request-limit-valid-1.yaml | pod-request-limit-valid-1 | cpu\\s+500m\\s+30 | memory\\s+(536870912\|512Mi)\\s+16Gi | # @case_id OCP-12049
      | ocp12145 | pod-request-limit-valid-2.yaml | pod-request-limit-valid-2 | cpu\\s+200m\\s+30 | memory\\s+(268435456\|256Mi)\\s+16Gi | # @case_id OCP-12145```

**Comment by jhou1** [comment]
Good catch @pruan-rht , I did not check scenario outlines, will do

**Comment by jhou1** [comment]
Now tested with `Scenario Outline`.

```
verification-tests ➤ tools/case_id_splitter.rb add-tags --tags @admin,@destructive,@gcp --cid OCP-12296   
Adding tag @admin to features/quickstarts/quickstarts.feature for test case OCP-12296
Adding tag @destructive to features/quickstarts/quickstarts.feature for test case OCP-12296
Adding tag @gcp to features/quickstarts/quickstarts.feature for test case OCP-12296
verification-tests ➤ tools/case_id_splitter.rb add-tags --tags @admin,@destructive,@gcp,@aws --cid OCP-12296
Test case OCP-12296 already has tag @admin
Test case OCP-12296 already has tag @destructive
Test case OCP-12296 already has tag @gcp
Adding tag @aws to features/quickstarts/quickstarts.feature for test case OCP-12296

git diff features/quickstarts/quickstarts.feature
diff --git a/features/quickstarts/quickstarts.feature b/features/quickstarts/quickstarts.feature
index 7f251d5c..b0e74558 100644
--- a/features/quickstarts/quickstarts.feature
+++ b/features/quickstarts/quickstarts.feature
@@ -1,6 +1,10 @@
 Feature: quickstarts.feature

   # @author haowang@redhat.com
+  @gcp
+  @destructive
+  @admin
+  @aws
   Scenario Outline: quickstart test
     Given I have a project
     When I run the :new_app client command with:
```

Tested with `Scenario`:
verification-tests ➤ tools/case_id_splitter.rb add-tags --tags @admin,@test --cid OCP-22578               
Adding tag @admin to features/tierN/apiserver/custom_resource.feature for test case OCP-22578
Adding tag @test to features/tierN/apiserver/custom_resource.feature for test case OCP-22578
verification-tests ➤ tools/case_id_splitter.rb add-tags --tags @admin,@test --cid OCP-22578               
Test case OCP-22578 already has tag @admin
Test case OCP-22578 already has tag @test

git diff apiserver/custom_resource.feature
diff --git a/apiserver/custom_resource.feature b/apiserver/custom_resource.feature
index 347636f29..91e2bcf91 100644
--- a/apiserver/custom_resource.feature
+++ b/apiserver/custom_resource.feature
@@ -2,6 +2,8 @@ Feature: API server related custom resource features

   # @author kewang@redhat.com
   # @case_id OCP-22578
+  @test
+  @admin
   Scenario Outline: Explain for CRD of config.openshift.io
     When I run the :explain client command with:
       | resource    | <item>                 |

**Comment by jhou1** [comment]
@liangxia @pruan-rht

File: tools/case_id_splitter.rb [hunk 1]
```diff
@@ -95,6 +95,22 @@ def run
         end
       end
 
+      command :"add-tags" do |c|
+        c.syntax = "#{$PROGRAM_NAME} add-tag [tag] [test_case]"
```
Comment (liangxia) [inline tools/case_id_splitter.rb:99]: Here is using `add-tag`, the others are using `add-tags`

Comment (jhou1) [reply tools/case_id_splitter.rb]: Thanks, corrected.

**Comment by pruan-rht** [comment]
/lgtm
@jhou1 for future enhancements, perhaps change options to put `--ids` to accept a set of cases instead of calling the script one case at a time...not a big deal.

**Comment by jhou1** [comment]
Thanks @pruan-rht I'll look into a future enhancement :)
/unhold

**Pull request merged**